scripts/login.js

```diff
@@ -0,0 +1,144 @@
+
+    const [passwordSpan, passwordConfirmSpan] =  selectPasswordSpan();
+    if(password.value !== password_confirm.value) {
+        password.classList.add('invalid');
+        password_confirm.classList.add('invalid');
+        passwordSpan.classList.remove('hide', 'valid');
+        passwordConfirmSpan.classList.remove('hide', 'valid');
+        passwordSpan.textContent = '비밀번호가 일치하지 않습니다.';
+        passwordConfirmSpan.textContent = '비밀번호가 일치하지 않습니다.';
+    } else if(password.value === password_confirm.value) {
+        password.classList.remove('invalid');
+        password_confirm.classList.remove('invalid');
+        passwordSpan.classList.add('hide', 'valid');
+        passwordConfirmSpan.classList.add('hide', 'valid');
+    }
+    checkValidity();
+}
+const passwordCheck = (e) => {
+    const [passwordSpan, passwordConfirmSpan] =  selectPasswordSpan();
+
+    if(password.value === "" || password.value.length < 8) {
+        password.classList.add('invalid');
+        passwordSpan.classList.remove('hide', 'valid');
+        passwordSpan.textContent = (password.value === "" ? '비밀번호를 입력해주세요.' : '비밀번호를 8자 이상 입력해주세요.');
+    } else if(password.value !== "" && password.value.length >= 8) {
+        password.classList.remove('invalid');
+        passwordSpan.classList.add('hide', 'valid');
+        if(password_confirm) passwordCompare();
+    }
+
+    if (password_confirm && e.target === password_confirm) {
+        if (password_confirm.value === "" || password_confirm.value.length < 8) {
+            password_confirm.classList.add('invalid');
+            passwordConfirmSpan.classList.remove('hide', 'valid');
+            passwordConfirmSpan.textContent = (password_confirm.value === "" ? '비밀번호를 입력해주세요.' : '비밀번호를 8자 이상 입력해주세요.');
+        } else if(password_confirm.value !== "" && password_confirm.value.length >= 8) {
+            password_confirm.classList.remove('invalid');
+            passwordConfirmSpan.classList.add('hide', 'valid');
+            passwordCompare();
+        }
+    }
+    checkValidity();
+}
+
+const nicknameConfirm = (e) => {
+    if(nickname.value === "") {
+        nickname.classList.add('invalid');
+        invalidSpanList[1].classList.remove('hide', 'valid');
+        invalidSpanList[1].textContent = '닉네임을 입력해주세요.';
+    } else if (e.target.value !== "") {
+        nickname.classList.remove('invalid');
+        invalidSpanList[1].classList.add('hide', 'valid');
+    }
+    checkValidity();
+}
+
+const showPassword = (e) => {
+    if(e.target === toggleSwitch[0]) {
+        if(passwordShowSwitch.checked) {
+            password.type = "password";
+        } else {
+            password.type = "text";
+        }
+    } else if(e.target === toggleSwitch[1]) {
+        if(passwordConfirmShowSwitch.checked) {
+            password_confirm.type = "password";
+        } else {
+            password_confirm.type = "text";
+        }
+    }
+}
+
+if(email) {
+    email.addEventListener('focusout', emailConfirm);
+}
```

Comment:
이런 경우 [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)을 활용하면 불필요한 if블락을 없애줄수있습니다 :)

```suggestion
    email?.addEventListener('focusout', emailConfirm);
```